main.py

```diff
@@ -298,169 +308,210 @@ def menu_live(params):
+        # content = None #string - current plugin content, e.g. ‘movies’ or ‘episodes’.
     )
 
+
 @common.plugin.action()
 def play_radio(params):
-    
-    cid = int(params.get('channel_id',None))
+    cid = int(params.get('channel_id', None))
     media_url = None
-    
+
     common.plugin.log("play_radio #{0}".format(cid))
 
     channel = api.get_single_channel(cid)
-    
+
     if channel:
         common.plugin.log(json.dumps(channel))
-        stream_node = channel.get('streamurl',None)
+        stream_node = channel.get('streamurl', None)
 
         if stream_node:
-            media_url = stream_node.get('mp3','').encode('utf-8').strip()
-        
+            media_url = stream_node.get('mp3', '').encode('utf-8').strip()
+
     if not media_url:
         common.plugin.log_error("unable to get stream URL.")
         common.popup("Impossible de trouver le flux media")
-        
-    #play
+
+    # play
     liz = xbmcgui.ListItem(path=media_url)
     return xbmcplugin.setResolvedUrl(handle=int(sys.argv[1]), succeeded=True, listitem=liz)
 
+
 @common.plugin.action()
 def play_media(params):
-    mid = int(params.get('media_id',None))
-    is_live = ( params.get('livemedia','') == 'True' )
-    drm = ( params.get('drm','') == 'True' )
-    
-    #common.popup("play media #{0} - live:{1} - drm:{2}".format(mid,is_live,drm))
-    common.plugin.log("play media #{0} - live:{1} - drm:{2}".format(mid,is_live,drm))
-    
-    #TOFIX DRM
-    if drm:
-        common.popup("Impossible actuellement de jouer des fichiers protégés par DRM.")
-        return False
-    
-    #get media details
-    media = api.get_media_details(mid,is_live)
+    # ------------uncomment to debug with pydevd ------------
+    # import sys
+    # sys.path.append('''C:\downloads\pycharm-debug.egg''')
+    #
+    # import pydevd
+    # pydevd.settrace('localhost', port=21000, stdoutToServer=True, stderrToServer=True)
+
+    mid = int(params.get('media_id', None))
+    is_live = (params.get('livemedia', '') == 'True')
+    drm = (params.get('drm', '') == 'True')
+
+    # common.popup("play media #{0} - live:{1} - drm:{2}".format(mid,is_live,drm))
+    common.plugin.log("play media #{0} - live:{1} - drm:{2}".format(mid, is_live, drm))
+
+    # # TOFIX DRM
+    # if drm:
+    #     common.popup("Impossible actuellement de jouer des fichiers protégés par DRM.")
+    #     return False
+
+    # get media details
+    media = api.get_media_details(mid, is_live)
     common.plugin.log(json.dumps(media))
-    
-    #get media stream URL
+
+    # get media stream URL
     media_url = None
     stream_node = media.get('url_streaming')
-    
+
     if stream_node:
-        #live media streaming
-        if is_live and utils.media_is_streaming(media):
-            media_url = stream_node.get('url_hls','').encode('utf-8').strip()
+        if drm:
+            media_url = api.get_video_detail(mid).get('data').get('urlHlsAes128', '').encode('utf-8').strip()
```

Comment:
Okay, so you use another API here, not the AUVIO one.
There should be a way to stick with the Auvio API; when reading the Android App code, I don't see any calls to the RTBF Api (or urlHlsAes128 property).
I don't understand why we can play this (_urlHlsAes128_) url and not the "regular" _url_hls_.

Anyway, a trick to get the same URL would be to use this, but it's kind of hackish:

        if drm:
            media_url = stream_node.get('url_hls','').encode('utf-8').strip()
            
            #TOFIX this URL does not work with current licence.
            #Update it so it match the property 'urlHlsAes128' at http://www.rtbf.be/api/media/video/?method=getVideoDetail&args[]=MEDIAIDHERE
            media_url = media_url.replace('/master.m3u8','-aes/master.m3u8')